app/models/spree/product_decorator.rb

```diff
@@ -1,3 +1,3 @@
 Spree::Product.class_eval do
-  has_many :page_view_events, -> { viewed.product }, class_name: 'Spree::PageEvent', foreign_key: :target_id
+  has_many :page_view_events, -> { where(activity: 'view') }, class_name: 'Spree::PageEvent', foreign_key: :target_id
```

Comment:
@harsh-ag Can you please move `where(activity: 'view')` this to a scope with view as a parameter

Reply:
used predefined scopes of cart_event and page_event in spree_events_tracker gem